maintainer/walberla_kernels/lees_edwards.py

```diff
@@ -59,16 +59,22 @@ def velocity_offset_eqs(config, method, pdfs, shear_dir_normal, stencil):
 
     grid_size = TypedSymbol("grid_size", dtype=default_dtype)
 
+    # When the kernels are combined, the integration loop also reaches
+    # one grid cell further to pull the populations. This changes the
+    # position where to apply the LE boundary by one cell.
+    boundary_offset = [0, grid_size - 1]
+    if (combined_kernel):
+        boundary_offset = [1, grid_size - 0]
     # +,-1 for upper/lower boundary layers, 0 otherwise.
```

Comment:
Please clarify: in the generated stream collide kernels based on the pull scheme, the loops in the generate kernels run on (1<=i<=n). For pure collisoin kernels they run on (0<=i<=n-1).